go/general_style_guide.md

```diff
@@ -86,7 +86,7 @@ A finding from `nancy` might look like:
 
 For each identified vulnerability:
 
-1. First, try to update the vulnerable dependency. If the dependency is not a _direct_ dependency, try using `go mod graph` to identify which package imports it, and update that instead.
+1. First, try to update the vulnerable dependency. If the dependency is not a _direct_ dependency, try using `go mod graph` to identify which package imports it, and update that instead. [helcaraxan/gomod](https://github.com/Helcaraxan/gomod) is also incredibly useful for debugging dependency issues. It can identify paths to a specific dependency with `gomod graph -d <your dependency>` and even represent the dependency tree graphically.
```

Comment:
Nice addition.

But `-d` flag no longer exists. What should we use now ?

Reply:
ooh, I didn't test it.
I just had a try at it, and Theo's right, the syntax changed and I can't manage to reverse-graph a dependency.

I things like this:
```
[prometheus-meta-operator]$ gomod.sh graph -p 'rdeps(github.com/giantswarm/microkit)'
08:52:40 warn query Empty query result. {"query": "github.com/giantswarm/microkit"}
strict digraph {
  node [shape=box,style="rounded,filled"]
  start=0
  splines=ortho
  "golang.org/x/text/internal/tag" [fontcolor="0.000 0.000 0.000",fillcolor="0.478 0.856 1.000"]
}
[prometheus-meta-operator]$ 
```

For info, gomod.sh runs gomod in a container:
```
#!/bin/bash

docker run --rm -it -w /app -v "$(pwd)":/app amstaffix/gomod:v0.7.1 "$@"
```


With docker image `v0.5.0` the "old" syntax works though:
```
[prometheus-meta-operator]$ gomod.sh graph -d github.com/giantswarm/microkit
strict digraph {
  node [shape=box,style=rounded]
  start=0
  splines=ortho
  "github.com/giantswarm/prometheus-meta-operator"
  "github.com/giantswarm/microkit"
  "github.com/giantswarm/exporterkit"
  "github.com/giantswarm/operatorkit/v7"
  "github.com/giantswarm/exporterkit" -> "github.com/giantswarm/microkit"
  "github.com/giantswarm/operatorkit/v7" -> "github.com/giantswarm/exporterkit"
  "github.com/giantswarm/prometheus-meta-operator" -> "github.com/giantswarm/exporterkit" [minlen=2]
  "github.com/giantswarm/prometheus-meta-operator" -> "github.com/giantswarm/microkit" [minlen=3]
  "github.com/giantswarm/prometheus-meta-operator" -> "github.com/giantswarm/operatorkit/v7"
}
[prometheus-meta-operator]$ 
```

Reply:
Let's please pin verison `v0.5.0` in here.

Reply:
Also adding the example above from Herve would help.

Reply:
I was able to get the old `-d` output using 0.7.1, so I've updated the example to just use the new syntax and mention the old 0.5.0 capability as a side note in case someone doesn't quite find what they need in 0.7.1+.

```
$ gomod graph 'rdeps(github.com/prometheus/client_golang)'
strict digraph {
  node [shape=box,style="rounded,filled"]
  start=0
  splines=ortho
  "github.com/aquasecurity/starboard" [fontcolor="0.000 0.000 1.000",fillcolor="0.624 0.813 1.000"]
  "github.com/giantswarm/starboard-exporter" [fontcolor="0.000 0.000 0.000",fillcolor="0.357 0.893 1.000"]
  "github.com/prometheus/client_golang" [fontcolor="0.000 0.000 0.000",fillcolor="0.102 0.969 1.000"]
  "github.com/prometheus/common" [fontcolor="0.000 0.000 0.000",fillcolor="0.384 0.885 1.000"]
  "go.etcd.io/etcd/client/v3" [fontcolor="0.000 0.000 0.000",fillcolor="0.208 0.358 1.000"]
  "go.etcd.io/etcd/server/v3" [fontcolor="0.000 0.000 0.000",fillcolor="0.298 0.340 1.000"]
  "k8s.io/apiextensions-apiserver" [fontcolor="0.000 0.000 1.000",fillcolor="0.753 0.774 1.000"]
  "k8s.io/apiserver" [fontcolor="0.000 0.000 0.000",fillcolor="0.373 0.325 1.000"]
  "k8s.io/component-base" [fontcolor="0.000 0.000 0.000",fillcolor="0.804 0.759 1.000"]
  "sigs.k8s.io/controller-runtime" [fontcolor="0.000 0.000 0.000",fillcolor="0.459 0.862 1.000"]
  "github.com/aquasecurity/starboard" -> "github.com/prometheus/client_golang" [minlen=4,style=dashed]
  "github.com/aquasecurity/starboard" -> "github.com/prometheus/common" [minlen=4,style=dashed]
  "github.com/aquasecurity/starboard" -> "k8s.io/apiextensions-apiserver" [minlen=2]
  "github.com/aquasecurity/starboard" -> "k8s.io/component-base" [minlen=3,style=dashed]
  "github.com/aquasecurity/starboard" -> "sigs.k8s.io/controller-runtime"
  "github.com/giantswarm/starboard-exporter" -> "github.com/aquasecurity/starboard"
  "github.com/giantswarm/starboard-exporter" -> "github.com/prometheus/client_golang" [minlen=5]
  "github.com/giantswarm/starboard-exporter" -> "github.com/prometheus/common" [minlen=5,style=dashed]
  "github.com/giantswarm/starboard-exporter" -> "k8s.io/apiextensions-apiserver" [minlen=3,style=dashed]
  "github.com/giantswarm/starboard-exporter" -> "k8s.io/component-base" [minlen=4,style=dashed]
  "github.com/giantswarm/starboard-exporter" -> "sigs.k8s.io/controller-runtime" [minlen=2]
  "github.com/prometheus/client_golang" -> "github.com/prometheus/common"
  "github.com/prometheus/common" -> "github.com/prometheus/client_golang"
  "go.etcd.io/etcd/client/v3" -> "github.com/prometheus/client_golang"
  "go.etcd.io/etcd/server/v3" -> "github.com/prometheus/client_golang" [minlen=2]
  "go.etcd.io/etcd/server/v3" -> "go.etcd.io/etcd/client/v3" [color=lightblue]
  "k8s.io/apiextensions-apiserver" -> "go.etcd.io/etcd/client/v3" [minlen=2,color=lightblue]
  "k8s.io/apiextensions-apiserver" -> "k8s.io/apiserver" [color=lightblue]
  "k8s.io/apiextensions-apiserver" -> "k8s.io/component-base" [minlen=2]
  "k8s.io/apiserver" -> "go.etcd.io/etcd/client/v3" [minlen=2,color=lightblue]
  "k8s.io/apiserver" -> "go.etcd.io/etcd/server/v3" [color=lightblue]
  "k8s.io/apiserver" -> "k8s.io/component-base"
  "k8s.io/component-base" -> "github.com/prometheus/client_golang" [minlen=2]
  "k8s.io/component-base" -> "github.com/prometheus/common" [minlen=2]
  "sigs.k8s.io/controller-runtime" -> "github.com/prometheus/client_golang" [minlen=4]
  "sigs.k8s.io/controller-runtime" -> "github.com/prometheus/common" [minlen=4,style=dashed]
  "sigs.k8s.io/controller-runtime" -> "k8s.io/apiextensions-apiserver"
  "sigs.k8s.io/controller-runtime" -> "k8s.io/component-base" [minlen=3]
}
```